### PR TITLE
modify type definition in headers.ts

### PR DESCRIPTION
**Pull request by qur786**
@maneetgoyal  Sir , I have updated some type definition , and I have already updated rest of the definition in object Dereference .

### CODE REVIEW

**Comment by maneetgoyal** [comment]
Will look after #33 is tackled

**Comment by maneetgoyal** [comment]
Lets try to fix this before #25

**Comment by qur786** [comment]
> Lets try to fix this before #25

Sir actually I am waiting a review from you .

**Comment by maneetgoyal** [comment]
There was merge conflict so didn't review. Will start in a bit now.

File: src/headers.ts [hunk 1]
```diff
@@ -592,8 +592,8 @@ export interface INCHIStereo0D {
 // }inchi_Input;
 
 export const inchi_Input = NAPIStructType({
-  atom: NAPIArrayType(inchi_Atom, 999),
-  stereo0D: NAPIArrayType(inchi_Stereo0D, 999),
+  atom: NAPIArrayType(inchi_Atom),
+  stereo0D: NAPIArrayType(inchi_Stereo0D),
```
Comment (maneetgoyal) [inline src/headers.ts:596]: Why is it reverted?

File: src/headers.ts [hunk 2]
```diff
@@ -875,8 +875,8 @@ export const inchi_InputEx = NAPIStructType({
   szOptions: refNAPI.types.CString,
   num_atoms: refNAPI.types.short,
   num_stereo0D: refNAPI.types.short,
-  polymer: refNAPI.refType(inchi_Input_Polymer),
-  v3000: refNAPI.refType(inchi_Input_V3000),
+  polymer: NAPIArrayType(inchi_Input_Polymer),
+  v3000: NAPIArrayType(inchi_Input_V3000),
```
Comment (maneetgoyal) [inline src/headers.ts:879]: Revert this change

File: src/headers.ts [hunk 3]
```diff
@@ -1735,8 +1735,8 @@ export const NORM_ATOM = NAPIStructType({
 // } NORM_ATOMS;
 
 export const NORM_ATOMS = NAPIStructType({
-  at: refNAPI.refType(NORM_ATOM),
-  at_fixed_bonds: refNAPI.refType(NORM_ATOM),
+  at: NAPIArrayType(NORM_ATOM),
+  at_fixed_bonds: NAPIArrayType(NORM_ATOM),
```
Comment (maneetgoyal) [inline src/headers.ts:1739]: You may need to revert this chnage, please check and confirm

File: tests/headers.test.ts [hunk 4]
```diff
@@ -229,8 +229,8 @@ describe("test inchi headers", () => {
     expect(inchiInputEx.szOptions).toBe("some-string");
     expect(inchiInputEx.num_atoms).toBe(11);
     expect(inchiInputEx.num_stereo0D).toBe(13);
-    expect(inchiInputEx.polymer.deref().n).toBe(0);
-    expect(inchiInputEx.v3000.deref().n_non_star_atoms).toBe(0);
+    expect(inchiInputEx.polymer[0].n).toBe(0);
+    expect(inchiInputEx.v3000[0].n_non_star_atoms).toBe(0);
```
Comment (maneetgoyal) [inline tests/headers.test.ts:233]: Revert the chnage

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: src/headers.ts [hunk 5]
```diff
@@ -110,7 +110,7 @@ export const ATOM_EL_LEN = 6; // length of ASCIIZ element symbol field
 export const NUM_H_ISOTOPES = 3; // number of hydrogen isotopes: protium, D, T
 export const ISOTOPIC_SHIFT_FLAG = 10000; // add to isotopic mass if isotopic_mass = (isotopic mass - average atomic mass)                             /* (isotopic mass - average atomic mass)        */
 export const ISOTOPIC_SHIFT_MAX = 100; // max abs(isotopic mass - average atomic mass)
-
+export const MAXARRAYSIZE = 999; // to create array with huge size
```
Comment (maneetgoyal) [inline src/headers.ts:113]:
```suggestion
export const MAX_ARRAY_SIZE = 999; // to create array with huge size
```

File: src/headers.ts [hunk 6]
```diff
@@ -1770,6 +1770,7 @@ export const NORM_ATOMS = NAPIStructType({
 export const INCHIGEN_DATA = NAPIStructType({
   pStrErrStruct: NAPIArrayType(refNAPI.types.char, STR_ERR_LEN),
   num_components: NAPIArrayType(refNAPI.types.int, INCHI_NUM),
+  // sir , look at below two key deifintion , should it be as it is or i have to change it to an array from pointer
```
Comment (maneetgoyal) [inline src/headers.ts:1773]:
```suggestion
```

Looks okay.

**Review by maneetgoyal — COMMENTED** [review]
Minor

**Comment by qur786** [comment]
there is some problem with stereo0D sir , in the last commit I have wrote some comments about it , please check it @maneetgoyal sir

**Review by maneetgoyal — APPROVED** [review]
👍

**Comment by maneetgoyal** [comment]
> there is some problem with stereo0D sir , in the last commit I have wrote some comments about it , please check it @maneetgoyal sir

@qur786 Should be fixed now

**Comment by qur786** [comment]
> > there is some problem with stereo0D sir , in the last commit I have wrote some comments about it , please check it @maneetgoyal sir
> 
> @qur786 Should be fixed now

yes sir

**Pull request merged**